packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFWebViewHostApi.m

```diff
@@ -0,0 +1,181 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FWFWebViewHostApi.h"
+#import <Flutter/Flutter.h>
+#import <WebKit/WebKit.h>
+#import "FWFDataConverters.h"
+
+@implementation FWFWebView
+- (void)setFrame:(CGRect)frame {
+  [super setFrame:frame];
+  // Prevents the contentInsets to be adjusted by iOS and gives control to Flutter.
+  self.scrollView.contentInset = UIEdgeInsetsZero;
+  if (@available(iOS 11, *)) {
+    // Above iOS 11, adjust contentInset to compensate the adjustedContentInset so the sum will
+    // always be 0.
+    if (UIEdgeInsetsEqualToEdgeInsets(self.scrollView.adjustedContentInset, UIEdgeInsetsZero)) {
+      return;
+    }
+    UIEdgeInsets insetToAdjust = self.scrollView.adjustedContentInset;
+    self.scrollView.contentInset = UIEdgeInsetsMake(-insetToAdjust.top, -insetToAdjust.left,
+                                                    -insetToAdjust.bottom, -insetToAdjust.right);
+  }
+}
+
+- (nonnull UIView *)view {
+  return self;
+}
+@end
+
+@interface FWFWebViewHostApiImpl ()
+@property FWFInstanceManager *instanceManager;
```

Comment:
`initWithInstanceManager` will be called by `FLTWebViewFlutterPlugin`, but I didn't plan on the plugin to manage it's lifecycle. `InstanceManager` was also made thread safe since multiple pigeon apis will be accessing and changing its contents. Should the property remain `atomic` since it's thread safe?

But now that I think about it, the `FlutterApi`s should use a `weak` reference. Otherwise there would be a reference loop. `InstanceManager`->`NavigationDelegate`->`CallbackMethod`->`FlutterApi`->`InstanceManager`.

Reply:
If `instanceManager` is thread safe, it can be nonatomic to enhance the performance.

Reply:
Yes, [FWFInstanceManager](https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFInstanceManager.m#L13) uses a `dispatch_queue_t` to manage access. `InstanceManager` will probably be accessed on different threads, but it's hard to tell. Objective-C delegate method documentation often excludes what thread it's being called on. So I just assume that it can and will be called on any thread.